tests/python/contrib/test_hexagon/metaschedule_e2e/test_resnet50_int8.py

```diff
@@ -359,11 +371,223 @@ def test_packed_8x8x32_resnet50(hexagon_launcher):
+            new_loops = sch.get_loops(block=conv2d_block)
+            sch.parallel(new_loops[2])
+            sch.unroll(new_loops[-4])
+            # TODO(nverke): Add compute optimizations here.
+        sch.blockize(loop=oc_i)
+
+        sch.tensorize(oc_i, VRMPY_u8i8i32_VTCM_INTRIN)
+
+        pipeline_loop = conv2d_loops[1]
+        sch.annotate(pipeline_loop, "software_pipeline_stage", [0, 0, 1, 2, 3])
+        sch.annotate(pipeline_loop, "software_pipeline_order", [0, 1, 2, 3, 4])
+        sch.annotate(pipeline_loop, "software_pipeline_async_stages", [0, 2])
+
+        return True
+
+    return schedule_fn
+
+
+def tune_async_dma_template(mod, params, hexagon_launcher):
+    """Generate async dma template."""
+
+    def schedule_rule_conv2d_async_dma(sch: Schedule, conv2d_block: BlockRV):
+        _schedule_async_dma_conv2d()(sch, conv2d_block)
+        return [sch]
+
+    register_func(
+        "meta_schedule.conv2d_NCHWc_int8.async_dma.hexagon", schedule_rule_conv2d_async_dma
+    )
+
+    def schedule_conv2d_for_tune(sch: Schedule):
+        _schedule_async_dma_conv2d()(sch)
+
+    # This line is necessary for link-params to take effect during
+    # task extraction and relay.build(...).
+    mod = mod.with_attr("executor", EXECUTOR)
+
+    with tempfile.TemporaryDirectory() as work_dir:
+        database = ms.relay_integration.tune_relay(
+            mod=mod,
+            target=TARGET_HEXAGON,
+            params=params,
+            work_dir=work_dir,
+            max_trials_global=20000,
+            max_trials_per_task=1,
+            num_trials_per_iter=1,
+            strategy="replay-trace",
+            builder=get_hexagon_local_builder(
+                tvm.transform.PassContext(
+                    opt_level=3,
+                    config={"tir.use_async_copy": 1, "tir.merge_async_commit_queue_scope": 0},
+                )
+            ),
+            runner=get_hexagon_rpc_runner(hexagon_launcher, number=20),
+            # Constrain search space to only be the single
+            # schedule provided for all blocks. No auto
+            # scheduling will be possible.
+            space=ms.space_generator.ScheduleFn(
+                schedule_conv2d_for_tune,
+                sch_rules=[],
+                postprocs=[],
+                mutator_probs={},
+            ),
+            # Without this, the same workloads with different constant weights
+            # are treated as distinct tuning tasks.
+            module_equality="ignore-ndarray",
+        )
+        return ms.relay_integration.compile_relay(
+            database=database,
+            mod=mod,
+            target=TARGET_HEXAGON,
+            params=params,
+            pass_config={
+                "tir.use_async_copy": 1,
+                "tir.merge_async_commit_queue_scope": False,
+            },
+        )
+
+
+@tvm.testing.requires_hexagon
+def test_async_dma_resnet50(hexagon_launcher):
```

Comment:
I think it should be possible to remove dups with `test_packed_8x8x32_resnet50`

Reply:
Was able to unify a good amount of the code ✅